### DISCUSSION TITLE
DATA_UPLOAD_MAX_MEMORY_SIZE exceeded

### DISCUSSION THREAD

**Discussion by gtondapu** [tethysplatform/tethys#643] [Q&A] [answered]
Hi all,

Can someone please help me resolve this issue? 

ERROR:django.security.RequestDataTooBig:Request body exceeded settings.DATA_UPLOAD_MAX_MEMORY_SIZE

From my tethys app, I am sending json to a request and it does not work because of it's size.  The size of json input varies according to different criteria.  Please let me know if you have questions.

**Comment by swainn** [comment]
Take a look at the documentation for the Django DATA_UPLOAD_MAX_MEMORY_SIZE setting: https://docs.djangoproject.com/en/2.2/ref/settings/#data-upload-max-memory-size

The documentation for the FILE_UPLOAD_MAX_MEMORY_SIZE may be helpful as well: https://docs.djangoproject.com/en/2.2/ref/settings/#std:setting-FILE_UPLOAD_MAX_MEMORY_SIZE

You set these settings in your portal_config.yml file in the settings section:

```yaml
settings:
  DATA_UPLOAD_MAX_MEMORY_SIZE: 123456789
```

You can check the setting using the `tethys settings` command:

```bash
tethys settings --get DATA_UPLOAD_MAX_MEMORY_SIZE
```

or if you want to see a list of all settings:

```bash
tethys settings --get
```
Here is the documentation on the settings command for reference: http://docs.tethysplatform.org/en/stable/tethys_cli/settings.html

**Comment by gtondapu** [comment, marked as answer]
I have tried this in portal_config.yml file inside .tethys folder.

tethys settings --get DATA_UPLOAD_MAX_MEMORY_SIZE 

This command does not reflect it

**Reply by swainn** [reply]
I just tested it Tethys 3.2.2 and it seemed to work. What version of Tethys are you running?

This is my portal_ config.yaml:

```yaml
# Portal Level Config File

# see tethys documentation for how to setup this file
version: 1.0
name:
apps: {}
settings:
  SECRET_KEY: ...
  DATA_UPLOAD_MAX_MEMORY_SIZE: 123456789
site_content: {}
```

If I run `tethys settings --get DATA_UPLOAD_MAX_MEMORY_SIZE` it gives me:

```
tethys settings --get DATA_UPLOAD_MAX_MEMORY_SIZ
DATA_UPLOAD_MAX_MEMORY_SIZE: 123456789
```

**Reply by swainn** [reply]
I forgot to mention that you can find the version of Tethys easily by running:

```
conda list tethys-platform
```

**Reply by gtondapu** [reply]
Mine is tethys 3.0.0

**Reply by swainn** [reply]
I believe there were some issues with settings in Tethys 3.0.0, that could be what is causing issues. [These issues were resolved in Tethys 3.1](http://docs.tethysplatform.org/en/stable/whats_new/prior_releases.html#tethys-portal-configuration). I'd recommend upgrading. There shouldn't be any breaking changes between 3.0.0 and the latest version 3.2.2.

Here's a link to our upgrade guide: http://docs.tethysplatform.org/en/stable/installation/update.html#update-tethys

**Reply by swainn** [reply]
Also, check out the What's New pages for 3.2 and 3.1 if you are curious what changes are included:

3.2: http://docs.tethysplatform.org/en/stable/whats_new.html#whats-new
3.1: http://docs.tethysplatform.org/en/stable/whats_new/prior_releases.html#release-3-1

**Reply by gtondapu** [reply]
Sure will look in to it Nathan! Thanks for suggesting. U just want confirm I am updating the correct yaml file. Can you please guide me to the yaml file path that should be updated?

**Reply by ezrajrice** [reply]
@gtondapu As of Tethys 3.0, the default location for the portal_config.yml file is in your $TETHYS_HOME/ directory. On a Linux machine the path would be something like `$ ~/.tethys/portal_config.yml`.

**Reply by gtondapu** [reply]
That worked when I added it outside settings block. Thank you both!

**Reply by swainn** [reply]
@gtondapu could you clarify what the fix was?

**Reply by gtondapu** [reply]
version: 1.0
name:
apps: {}
DATA_UPLOAD_MAX_MEMORY_SIZE: 123456789
settings:
  SECRET_KEY: ...
site_content: {}

This was the fix. It worked when I added the variable outside the settings block in portal_ config.yaml.

**Reply by ezrajrice** [reply]
> 
> 
> version: 1.0
> name:
> apps: {}
> DATA_UPLOAD_MAX_MEMORY_SIZE: 123456789
> settings:
> SECRET_KEY: ...
> site_content: {}
> 
> This was the fix. It worked when I added the variable outside the settings block in portal_ config.yaml.

@gtondapu  I'm glad that worked for you. For future reference, this worked because the non-changeable settings.py file expected this format for version 3.0. Starting in Tethys version 3.1, this was changed for consistency that all Django settings be listed inside the settings section, see [Tethys Portal Configuration Docs](http://docs.tethysplatform.org/en/stable/tethys_portal/configuration.html) If you decide to update to a newer version, you will need to modify your portal_config.yml file to reflect the new format as shown in Nathan's first response.

Example:
> ```yaml
> # Portal Level Config File
> 
> # see tethys documentation for how to setup this file
> version: 1.0
> name:
> apps: {}
> settings:
>   SECRET_KEY: ...
>   DATA_UPLOAD_MAX_MEMORY_SIZE: 123456789
> site_content: {}
> ```

**Reply by swainn** [reply]
+1 to @ezrajrice's comment. The reason your solution worked @gtondapu is because of a bug in Tethys 3.0. I really recommend that you upgrade to at least Tethys 3.1 to avoid headaches in the future.
